doc/migrations/v0.46-v1.0.0.md

```diff
@@ -1,16 +1,101 @@
+<!--Specify versions for migration below-->
 # Migrating to libp2p@1.0.0 <!-- omit in toc -->
 
-A migration guide for refactoring your application code from libp2p `v0.46` to `v1.0.0`.
+A migration guide for refactoring your application code from libp2p v0.46 to v1.0.0.
 
 ## Table of Contents <!-- omit in toc -->
 
-- [New features](#new-features)
-- [Breaking changes](#breaking-changes)
+- [API](#api)
+- [Module Updates](#module-updates)
+- [Metrics](#metrics)
 
-## New features
+## API
 
-...
+<!--Describe breaking APIs with examples for Before and After
+Example:
 
-## Breaking changes
+### Peer Discovery
 
-...
+__Describe__
+
+**Before**
+
+```js
+
+```
+
+**After**
+
+```js
+
+```
+
+-->
+
+### KeyChain
+
+The KeyChain object is no longer included on Libp2p and must be instantiated explicitly if desired.
+
+**Before**
+
+```ts
+import type { KeyChain } from '@libp2p/interface/keychain'
+
+const libp2p = await createLibp2p(...)
+
+const keychain: KeyChain = libp2p.keychain
+```
+
+**After**
+
+```ts
+import { DefaultKeyChain } from '@libp2p/keychain'
+import type { KeyChain } from '@libp2p/interface/keychain'
+
+const libp2p = await createLibp2p({
+  ...
+  services: {
+    keychain: (components) => new DefaultKeyChain(components, {
+      ...DefaultKeyChain.generateOptions()
+    })
+  }
+})
+
+const keychain: KeyChain = libp2p.services.keychain
+```
+
+### AutoNAT
```

Comment:
```suggestion
## AutoNAT
```

Reply:
These are intentionally `###` to be under `## API`